scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -1006,12 +1006,12 @@ def _create_build_dir_link(self, links_dir_path, instance):
 
 def change_skip_to_error_if_integration(options, instance):
     ''' All skips on integration_platforms are treated as errors.'''
-    if instance.platform.name in instance.testsuite.integration_platforms \
-        and "quarantine" not in instance.reason.lower():
-        # Do not treat this as error for a list of filter types.
+    if instance.platform.name in instance.testsuite.integration_platforms:
+        # Do not treat this as error if filter type is among ignore_filters
         filters = {t['type'] for t in instance.filters}
         ignore_filters ={Filters.CMD_LINE, Filters.SKIP, Filters.PLATFORM_KEY,
```

Comment:
It depends what is the presumption and logic behind. As I understand, by **default** we assume, that a config for a platform under integration_platforms **must work**. Therefore, skips are turned to errors on it by default with exception for clearly stated reasons. The current logic follows this paradigm. Doing the filtering in revers, as suggested in the comment, would make the code shorter, but won't follow the paradigm. It could result with new filters being added but not included in the list for "skip to error" logic, hence hinder some real issues/regressions.
 E.g. this PR introduces new filter but doesn't address the "skip to error" for integration platforms https://github.com/zephyrproject-rtos/zephyr/pull/66419. In that case, the new filter should also be added to the exception list. A committer might be unaware of the paradigm, and don't add an exception. In such case it is IMO better to figure out about it later on, with some easy-to-fix false positive (".. skipped but is on the integration list"), than (with reversed logic) spend more time figuring out why some regression leak in later on, even though we had a platform under integration_platforms. 
 If you don't agree with my reasoning, I can do it the "reversed" way.